.github/workflows/build-release.yml

```diff
@@ -0,0 +1,46 @@
+name: build
+on:
+  pull_request:
+  # Run builds nightly to catch incompatibilities with new marshmallow releases
+  schedule:
+  - cron: "0 0 * * *"
+jobs:
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - run: python -m pip install --upgrade pip wheel
+      - run: pip install tox
+      - run: tox -elint
+  tests:
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - {name: '3.7-ma3', python: '3.7', os: ubuntu-latest, tox: py37-marshmallow3}
+          - {name: '3.10-ma3', python: '3.10', os: ubuntu-latest, tox: py310-marshmallow3}
+          - {name: '3.10-madev', python: '3.10', os: ubuntu-latest, tox: py310-marshmallowdev}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - run: python -m pip install --upgrade pip wheel
+      - run: pip install tox
+      - run: tox -e${{ matrix.tox }}
+  release:
+    needs: [lint, tests]
+    name: PyPI release
+    if: startsWith(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - run: python -m pip install --upgrade pip wheel
+      - run: pip install twine
+      - run: python setup.py sdist bdist_wheel
```

Comment:
the pypa docs recommend using `build` rather than setup.py: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

Reply:
Thanks for the link. I shall look into this. Perhaps it didn't exist back when I set this up for flask-smorest. Or I copied code from an outdated blog post.

Reply:
I just changed this, copying this part from webargs. We now use build as instructed in the docs. We're still using twine for consistency with webargs, instead of the GH action proposed in the docs.